packages/@tinacms/toolkit/src/packages/fields/components/Reference.tsx

```diff
@@ -0,0 +1,185 @@
+/**
+
+Copyright 2021 Forestry.io Holdings, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+import * as React from 'react'
+import styled from 'styled-components'
+import { useCMS } from '../../../react-tinacms/use-cms'
+import { TinaCMS } from '../../../tina-cms'
+
+type Option = {
+  value: string
+  label: string
+}
+
+interface ReferenceFieldProps {
+  label?: string
+  name: string
+  component: string
+  options: (Option | string)[]
+}
+
+export interface ReferenceProps {
+  name: string
+  input: any
+  field: ReferenceFieldProps
+  disabled?: boolean
+  options?: (Option | string)[]
+}
+
+const useGetNode = (cms: TinaCMS, id: string) => {
+  const [document, setDocument] = React.useState(undefined)
+
+  React.useEffect(() => {
+    const fetchNode = async () => {
+      const response = await cms.api.tina.request(
+        `
+        query($id: String!) {
+          node(id:$id) {
+            ... on Document {
+              sys {
+                collection {
+                  name
+                }
+                breadcrumbs
+              }
+            }
+          }
+        }`,
+        { variables: { id } }
+      )
+
+      setDocument(response.node)
+    }
+    if (cms && id) {
+      fetchNode()
+    } else {
+      setDocument(undefined)
+    }
+  }, [cms, id])
+  return document
+}
+
+const GetReference = ({ cms, id, children }) => {
+  const document = useGetNode(cms, id)
+
+  if (!document) {
+    return null
+  }
+  return <>{children(document)}</>
+}
+
+export const Reference: React.FC<ReferenceProps> = ({
+  input,
+  field,
+  options,
+}) => {
+  const cms = useCMS()
+  const hasTinaAdmin = cms.flags.get('tina-admin')
+
+  const selectOptions = options || field.options
+  return (
+    <SelectElement>
+      <select
+        id={input.name}
+        value={input.value}
+        onChange={input.onChange}
+        {...input}
+      >
+        {selectOptions ? (
+          selectOptions.map(toProps).map(toComponent)
+        ) : (
+          <option>{input.value}</option>
+        )}
+      </select>
+      {hasTinaAdmin && (
+        <GetReference cms={cms} id={input.value}>
+          {(document) => (
+            <a
+              href={`/admin/collections/${
+                document.sys.collection.name
+              }/${document.sys.breadcrumbs.join('/')}`}
+            >
+              &raquo; Edit in TinaAdmin
+            </a>
+          )}
+        </GetReference>
```

Comment:
Used a render-prop pattern similar to what `TinaAdmin` uses.  `document` is the response from a `query { node }` request, so we have access to all of the `sys` information for the Document.

When the `tina-admin` flag is disabled, this field behaves just like the `select` one.